sprint-mission-7/queries.sql

```diff
@@ -0,0 +1,172 @@
+/*
+  다음 경우들에 대해 총 14개의 SQL 쿼리를 작성해 주세요.
+  예시로 값이 필요한 경우 적당한 값으로 채워넣어서 작성하면 됩니다. 
+*/
+
+/*
+  1. 내 정보 업데이트 하기
+  - 닉네임을 "test"로 업데이트
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+UPDATE users
+SET nickname = 'test'
+WHERE id = 1;
+
+
+/*
+  2. 내가 생성한 상품 조회
+  - 현재 로그인한 유저 id가 1이라고 가정
+  - 최신 순으로 정렬
+  - 10개씩 페이지네이션, 3번째 페이지
+*/
+SELECT *
+FROM products
+WHERE user_id = 1 
+ORDER BY created_at DESC
+LIMIT 10 OFFSET 20;
+
+/*
+  3. 내가 생성한 상품의 총 개수
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+
+SELECT 
+  COUNT(*) AS created
+FROM products
+WHERE user_id = 1;
+
+
+/*
+  4. 내가 좋아요 누른 상품 조회
+  - 현재 로그인한 유저 id가 1이라고 가정
+  - 최신 순으로 정렬
+  - 10개씩 페이지네이션, 3번째 페이지
+*/
+
+SELECT *
+FROM product_likes
+WHERE user_id = 1 
+ORDER BY created_at DESC
+LIMIT 10 OFFSET 20;
+
+
+/*
+  5. 내가 좋아요 누른 상품의 총 개수
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+
+SELECT 
+  COUNT(*) AS created
+FROM product_likes
+WHERE user_id = 1;
+
+
+/*
+  6. 상품 생성
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+
+INSERT INTO products (user_id, image, name, description, price)
+VALUES (1,'img_rainbow_doll.jpg','무지개 곰인형','파스텔 무지개톤의 귀여운 곰인형',39000);
+
+SELECT * FROM products WHERE name = '무지개 곰인형';
+
+/*
+  7. 상품 목록 조회
+  - 상품명에 "test"가 포함된 상품 검색
+  - 최신 순으로 정렬
+  - 10개씩 페이지네이션, 1번째 페이지
+  - 각 상품의 좋아요 개수를 포함해서 조회하기
+*/
+
+-- 계산식으로 likeCount 가져오기 위해 
+-- DB의 컬럼 삭제
+ALTER TABLE products
+DROP COLUMN like_count;
+
+ALTER TABLE articles
+DROP COLUMN like_count;
+
+SELECT 
+  products.id,
+  products.name,
+  products.description,
+  Count(product_likes.user_id) total_like,
+  products.created_at
+FROM products JOIN product_likes
+ON products.id = product_likes.product_id
+GROUP BY products.id
+HAVING products.name LIKE 'test%'
```

Comment:
```suggestion
HAVING products.name LIKE '%test%'
```
포함 조건이기 때문에 양방향이여야 할 것 같아욤!